hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/search/TermHelperTest.java

```diff
@@ -0,0 +1,41 @@
+package ca.uhn.fhir.jpa.dao.search;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class TermHelperTest {
+
+	@Test
+	void empty_returns_empty() {
+		assertEquals( Collections.emptySet(), TermHelper.makePrefixSearchTerm(Collections.emptySet()) );
+	}
+
+	@Test
+	void noQuotedSpcedOrStarElements_return_star_suffixed() {
+		Set<String> result = TermHelper.makePrefixSearchTerm(Set.of("abc", "def", "ghi"));
+		assertEquals( Set.of("abc*", "def*", "ghi*"), result );
+	}
+
```

Comment:
I'm not sure about all the results here, but we need to handle multi-term queries.
```suggestion
	@Test
 	void multiSimpleTerm_hasSimpleTermsWildcarded() {
 		Set<String> result = TermHelper.makePrefixSearchTerm(Set.of("abc def"));
 		assertEquals( Set.of("abc* def*"), result );
 	}

  // question - do we want to wildcard simple terms inside other syntax for consistency?
	@Test
 	void simpleQuerySytax_hasSimpleTermsWildcarded() {
 		Set<String> result = TermHelper.makePrefixSearchTerm(Set.of("(def | efg)", "ghi +(\"abc" \"def\")"));
 		assertEquals( Set.of("(def* | efg*)", "ghi* +(\"abc" \"def\")"), result );
 	}

```